python/lsst/pipe/tasks/fit_multiband.py

```diff
@@ -0,0 +1,314 @@
+        catexps : `typing.List [CatalogExposure]`
+            A list of catalog-exposure pairs in a given band.
+        cat_ref : `lsst.afw.table.SourceCatalog`
+            A reference source catalog to fit.
+
+        Returns
+        -------
+        retStruct : `lsst.pipe.base.Struct`
+            A struct with a cat_output attribute containing the output
+            measurement catalog.
+
+        Notes
+        -----
+        Subclasses may have further requirements on the input parameters,
+        including:
+            - Passing only one catexp per band;
+            - Catalogs containing HeavyFootprints with deblended images;
+            - Fitting only a subset of the sources.
+        If any requirements are not met, the subtask should fail as soon as
+        possible.
+        """
+        raise NotImplementedError()
+
+    @property
+    @abstractmethod
+    def schema(self) -> afwTable.Schema:
+        raise NotImplementedError()
+
+
+class MultibandFitConfig(
+    pipeBase.PipelineTaskConfig,
+    pipelineConnections=MultibandFitConnections,
+):
+    """ Configuration class for the MultibandFitTask, containing a
+    configurable subtask that does all fitting.
+    """
+    fit_multiband = pexConfig.ConfigurableField(
+        target=MultibandFitSubTask,
+        doc="Task to fit sources using multiple bands",
+    )
+
+    def get_band_sets(self):
+        """ Get the set of bands required by the fit_multiband subtask.
+
+        Returns
+        -------
+        bands_fit : `set`
+            The set of bands that the subtask will fit.
+        bands_read_only : `set`
+            The set of bands that the subtask will only read data (measurement catalog and exposure) for.
+        """
+        try:
+            bands_fit = self.fit_multiband.bands_fit
+        except AttributeError:
+            raise RuntimeError(f'{__class__}.fit_multiband must have bands_fit attribute') from None
+        bands_read_only = self.fit_multiband.bands_read_only()
+        return set(bands_fit), set(bands_read_only)
+
+
+class MultibandFitTask(pipeBase.PipelineTask):
+    ConfigClass = MultibandFitConfig
+    _DefaultName = "multibandFit"
+
+    def __init__(self, initInputs, **kwargs):
+        super().__init__(initInputs=initInputs, **kwargs)
+        self.makeSubtask("fit_multiband", schema=initInputs["cat_ref_schema"].schema)
+        self.cat_output_schema = afwTable.SourceCatalog(self.fit_multiband.schema)
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+        input_refs_objs = [(inputRefs.cats_meas, inputs['cats_meas']), (inputRefs.coadds, inputs['coadds'])]
+        cats, exps = [
+            {dRef.dataId: obj for dRef, obj in zip(refs, objs)}
+            for refs, objs in input_refs_objs
+        ]
+        dataIds = set(cats).union(set(exps))
+        catexps = [
+            CatalogExposure(catalog=cats.get(dataId), exposure=exps.get(dataId), dataId=dataId)
```

Comment:
Now that CatalogExposures' catalogs and exposures are optional (I can see future use cases for just wanting one or the other attached to a dataId), I went with defensively putting inputs into dicts as you suggested and in case future changes to valid subtasks decide to drop one or the other. Does that seem reasonable?